cmd/protoc-gen-gorums/gengorums/template_server.go

```diff
@@ -23,22 +27,23 @@ var registerInterface = `
 {{$service := .GoName}}
 func (s *GorumsServer) Register{{$service}}Server(srv {{$service}}) {
 	{{- range nodeStreamMethods .Methods}}
-	s.srv.handlers[{{unexport .GoName}}MethodID] = func(in *gorumsMessage ,{{if isOneway .}} _ {{- else}} finished {{- end}} chan<- *gorumsMessage) {
+	s.srv.handlers[{{unexport .GoName}}MethodID] = func(ctx {{$context}}, in *gorumsMessage ,{{if isOneway .}} _ {{- else}} finished {{- end}} chan<- *gorumsMessage) {
 		req := in.message.(*{{in $genFile .}})
 		{{- if isOneway .}}
-		srv.{{.GoName}}(req)
+		srv.{{.GoName}}(ctx, req)
 		{{- else }}
-		c := make(chan *{{out $genFile .}})
-		go func() {
-			resp := <-c
-			finished <- &gorumsMessage{metadata: in.metadata, message: resp}
-		}()
-		srv.{{.GoName}}(req, c)
+		once := new({{use "sync.Once" $genFile}})
```

Comment:
Could you add a brief comment here explaining why we use the `once.Do()` thing?

Also, does the `finished` channel mean the queue of messages to be sent on the stream? I think here it sounds okay to use `finished` (maybe), but in the `NodeStream` implementation, it is also called finished. Maybe this should be called something else? (outQueue and sendQueue?)